onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1608,9 +1609,23 @@ bool GemmOpBuilder::IsQuantizedOp(const NodeUnit& node_unit) const {
 void GemmOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const NodeUnit& node_unit) const {
   const auto& inputs = node_unit.Inputs();
   if (IsQuantizedOp(node_unit)) {
-    AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[0].quant_param);               // b_scale, b_zp
-    AddInputToSkip(model_builder, inputs[1]);                                                    // b, b_scale, b_zp
-    AddQuantizationScaleAndZeroPointToSkip(model_builder, *node_unit.Outputs()[0].quant_param);  // y_scale, y_zp
+    if (node_unit.OpType() == "QlinearMatMul") {
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[0].quant_param);               // a_scale, a_zp
+      AddInputToSkip(model_builder, inputs[1]);                                                    // b, b_scale, b_zp
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *node_unit.Outputs()[0].quant_param);  // y_scale, y_zp
+    }
+    if (node_unit.OpType() == "Gemm") {                                               // QDQGemm
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[0].quant_param);  // x_scale, x_zp
+      AddQuantizationScaleAndZeroPointToSkip(model_builder, *inputs[1].quant_param);  // w_scale, w_zp
+      NodeAttrHelper helper(node_unit);
+      const auto transB = helper.Get("transB", 0);
+      if (transB == 0)
+        model_builder.AddInitializerToSkip(inputs[1].node_arg.Name());
+      if (inputs.size() > 2) {
```

Comment:
What if transB is 1? Do we handle that elsewhere? If so a comment here capturing that background info would be good.

Reply:
I guess the initializer - input B only needs to be skipped for the GEMM (not transpose) case as we will handle the initializer B later (add the initializer and transpose it as nnapi only supports `A*B^T`) via `AddInitializerTransposed()` this method https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc#L238-L241

will add comments for this.